src/equations/compressible_euler_2d.jl

```diff
@@ -705,11 +707,17 @@ function signature with argument `which` set to `Val{:minus}()` or `Val{:plus}()
   Flux Vector Splitting of the Inviscid Gasdynamic Equations
   With Application to Finite Difference Methods
   [NASA Technical Memorandum](https://ntrs.nasa.gov/api/citations/19790020779/downloads/19790020779.pdf)
+- D. Drikakis and S. Tsangaris (1993)
+  On the solution of the compressible Navier-Stokes equations using
+  improved flux vector splitting methods
+  [DOI: 10.1016/0307-904X(93)90054-K](https://doi.org/10.1016/0307-904X(93)90054-K)
```

Comment:
I'm not sure whether we should still call the curved version `splitting_steger_warming` since it does not reduce to the `splitting_steger_warming` we have implemented for the Cartesian case, does it?

```julia
julia> let
       equations = CompressibleEulerEquations2D(1.4)
       u = SVector(1.0, -0.2, 0.3, 1.0)
       cartesian = @show splitting_steger_warming(u, Val(:plus), 1, equations)
       curved = @show splitting_steger_warming(u, Val(:plus), SVector(1.0, 0.0), equations)
       cartesian - curved
       end
splitting_steger_warming(u, Val(:plus), 1, equations) = [0.18700075021607912, 0.09791398562785399, 0.05610022506482373, 0.22987620366267872]
splitting_steger_warming(u, Val(:plus), SVector(1.0, 0.0), equations) = [0.2618010503025108, 0.08295392561056766, 0.07854031509075322, 0.35971464311564977]
4-element SVector{4, Float64} with indices SOneTo(4):
 -0.07480030008643165
  0.014960060017286322
 -0.02244009002592949
 -0.12983843945297105
```

Could it be better to call it `splitting_drikakis_tsangris` instead?

Reply:
No, this version does not reduce to the normal `splitting_steger_warming` in the Cartesian directions because of some manipulations done by Drikakis and Tsangris, e.g., they eliminate the need for one of the eigenvalues from the original Steger-Warming. 

I went back and forth about what to name this function, because the directly curvilinear extension of Steger-Warming does not work. The critical relationship the curvilinear flux splitting needs is that
```
flux(u, normal_direction, equations) 
  - splitting(u, Val{:plus), normal_direction, equations)
  - splitting(u,Val{:minus}, normal_direction, equations)
```
returns a vector with only round-off errors. A straightforward extension of the Cartesian Steger-Warming does not satisfy this condition (among others I have tried). Only the version I found from Drikakis and Tsangris actually worked.

So, in short, I am open to changing the name. I just need to write good docstrings as to why :)

Reply:
I would prefer a `splitting_drikakis_tsangaris` - that we could implement also for the Cartesian case. Then, we should mention that `splitting_drikakis_tsangaris` should be used as improved alternative on curved meshes in the docstring of `splitting_steger_warming`.

Reply:
That is a good idea. I can implement the Cartesian version of Drikakis and Tsangris. To be doubly sure I re-implemented to version from the Steger-Warming paper and it converges but is not free-stream. The issue is that some of the terms have a higher power dependency on the metric terms which would require sixth order boundary closures to guarantee FSP.

Reply:
I added the Cartesian counterpart with a corresponding test. I also included a reference to the new `splitting_drikakis_tsangaris` in the `splitting_steger_warming` docstring.